bundles/org.openhab.binding.hdmicec/src/main/java/org/openhab/binding/hdmicec/internal/HdmiCecEquipmentHandler.java

```diff
@@ -0,0 +1,233 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hdmicec.internal;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.regex.Matcher;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HdmiCecEquipmentHandler} is responsible for handling commands,
+ * which are sent to one of the channels.
+ *
+ * @author David Masshardt - Initial contribution
+ * @author Sam Spencer - Discovery, Conversion to OH3 and submission
+ */
+@NonNullByDefault
+public class HdmiCecEquipmentHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HdmiCecEquipmentHandler.class);
+
+    private @Nullable HdmiCecBindingConfiguration config;
+
+    // config paramaters
+    private String deviceIndex = "unknown"; // hex number, like 0 or e
+    private String address = "unkonwn"; // of the form 0.0.0.0
+
+    private @Nullable HdmiCecBridgeHandler bridgeHandler;
+    private @Nullable TimerWrapper myTimer = null;
+
+    private ThingUID uid;
+    private @Nullable Bridge bridge = null;
+
+    public HdmiCecEquipmentHandler(Thing thing) {
+        super(thing);
+        uid = thing.getUID();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            bridge = getBridge();
+            config = getConfigAs(HdmiCecBindingConfiguration.class);
+
+            if (config != null) {
+                if (config.deviceIndex != null) {
+                    deviceIndex = "" + config.deviceIndex;
+                }
+                if (config.address != null) {
+                    address = "" + config.address;
+                }
+            }
+            thing.setLabel(thing.getLabel().replace("Equipment", uid.getId()));
+
+            logger.debug("Initializing thing {}", uid);
+            bridgeHandler = (HdmiCecBridgeHandler) bridge.getHandler();
+
+            if (bridge.getStatus() == ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } catch (Exception e) {
+            logger.error("Error in initialize: {} at {}", e.toString(), e.getStackTrace());
```

Comment:
use the implicit last parameter to `logger.error()` to log exceptions (also applies to other logger calls)